spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/zuul/filters/SimpleRouteLocator.java

```diff
@@ -94,14 +87,19 @@ public Route getMatchingRoute(final String path) {
 
 	}
 
+	protected Map<String, ZuulRoute> getRoutesMap() {
+		if (this.routes.get() == null) {
+			this.routes.set(locateRoutes());
+		}
+		return Collections.unmodifiableMap(this.routes.get());
```

Comment:
Hi Spencer,

I'm not sure what you mean. I introduced the unmodifiableMap in getRoutes() in my prior PR commit that used the assembleRoutes callback. Since that was a no go, I moved it to the getRoutesMap() method in this PR commit so that subclasses won't be able to modify the ZuulRoutes map.